apiserver/params/applications.go

```diff
@@ -554,3 +554,21 @@ type UnitInfoResult struct {
 type UnitInfoResults struct {
 	Results []UnitInfoResult `json:"results"`
 }
+
+// ExposeInfoResults the expose info for a list of applications.
+type ExposeInfoResults struct {
+	Results []ExposeInfoResult `json:"results"`
+}
+
+// ExposeInfoResult holds the result of a GetExposeInfo call.
+type ExposeInfoResult struct {
+	Error *Error `json:"error,omitempty"`
+
+	Exposed bool `json:"exposed,omitempty"`
+
+	// Expose parameters grouped by endpoint name. An empty ("") endpoint
```

Comment:
Would it benefit to use "*" as the match everything string? For clarity when doing db dumps etc?
Maybe not worth it. But if not already done, I assume there's a const defined?

Reply:
The `""` was chosen so we are consistent with endpoint bindings (the `""` represents the default space for all endpoints) both in persisted state and in bundles. Also, the same pattern is used in the openedPorts collection to indicate that a port is opened for all endpoints (present and future -- see https://github.com/juju/juju/pull/11863 for more details about the wildcard endpoint wrt. port ranges). 

Most of the tests use the `allEndpoints` const but the logic typically checks for `""` (though I have sprinkled comments around those locations to explain what this represents)

Reply:
I wish we had used "*" for the "default binding" in the beginning. It would make things much easier to talk about.
I wonder if we could sneak it into a 3.0 ...